pkg/attest/attest.go

```diff
@@ -87,11 +88,39 @@ func GenerateMAAHostData(inputBytes []byte) [HOST_DATA_SIZE]byte {
 //	retrieval and has been reported by the PSP in the attestation report as REPORT DATA
 //
 // Note that it uses fake attestation report if it's not running inside SNP VM
+
 func (certState *CertState) Attest(maa common.MAA, runtimeDataBytes []byte, uvmInformation common.UvmInformation) (string, error) {
+
+	SNPReportBytes, vcekCertChain, inittimeDataBytes, uvmReferenceInfoBytes, err := certState.getCollateral(maa, runtimeDataBytes, uvmInformation)
+	if (err != nil) {
+		return "", err
+	}
+	// Retrieve the MAA token required by the request's MAA endpoint
+	logrus.Info("Retrieving MAA token...")
+	maaToken, err := maa.Attest(SNPReportBytes, vcekCertChain, inittimeDataBytes, runtimeDataBytes, uvmReferenceInfoBytes)
+	if err != nil || maaToken == "" {
+		return "", errors.Wrapf(err, "Retrieving MAA token from MAA endpoint failed")

```

Comment:
If maaToken is empty and err == nil, does the errors.Wrapf error out?

Reply:
Hmm, pre-existing [bug](https://github.com/microsoft/confidential-sidecar-containers/blob/9d770388542fc0b5562e86d7b1df7d81a51d1aec/pkg/attest/attest.go#L183) but I think it happens that that can never be the case due to https://github.com/microsoft/confidential-sidecar-containers/blob/9d770388542fc0b5562e86d7b1df7d81a51d1aec/pkg/common/maa.go#L195. I might fix it though.